src/inventory/transaction/action/SlotChangeAction.php

```diff
@@ -74,6 +76,17 @@ public function validate(Player $source) : void{
 		if($this->targetItem->getCount() > $this->inventory->getMaxStackSize()){
 			throw new TransactionValidationException("Target item exceeds inventory max stack size");
 		}
+		if($this->inventory instanceof SlotSafeInventory && !$this->targetItem->equals(VanillaItems::AIR())){
+			$result = true;
+			foreach($this->inventory->getSlotValidators() as $validator){
+				if(($result = $validator($this->inventory, $this->targetItem, $this->inventorySlot))){
```

Comment:
See above about logic inversion

Reply:
Thats currently the behavior. all validator must accept the transaction, however it will be rejected.